### ISSUE TITLE
Request: Bring back the --jgi option

### ISSUE THREAD

**Issue by GabeAl** [RasmussenLab/vamb#355]
TL;DR: I want to run this on nearly 100,000 metagenomes for single-sample binning, and need the process to be as time- and space- optimized as possible.  I just want to supply my existing high-quality contig depth estimates in jgi or tsv formats. 

I came across the AAMB paper and read through the original AAMB fork (not realizing it had been merged back and called V4 of VAMB), and decided I would like to try VAMB/AAMB within context of my existing workflow. I like metabat2 because it is a single statically compilable binary that is portable, in a static megahit + static aligner + static metabat2 workflow. 

I have solved the annoying dependency on other aligners by making an aligner that outputs results directly in the form output by jgi_summarize, and it is an optimal aligner that resolves multi-mapping reads perfectly by laying down the equivalent of 1/N coverage of that read onto each genome where it aligns (a generalization of the "random tiebreaking" expected by metabat2 that works better with lower-abundance contigs).  That means no huge bam files, no space or parsing time overhead at all, just 3 programs that directly work with each other much faster, no hisat dependency, no sorting, nothing. Just quick, simple, and portable. Other newer aligners are doing [similarly](https://github.com/ksahlin/strobealign#abundance-estimation-mode-for-metagenomic-binning). That said, I am happy to relax my no-external-dependency requirements to try VAMB for single-sample binning (yes, single sample -- I know this wasn't a focus area for VAMB, but at the scale we're talking about, it makes a lot of sense), by making a conda environment to handle the major dependencies and pip installing into it. 

I then came across a very surprising change mentioned in 4.0 where, if I'm understanding correctly, in order to supply contig depths, I have to enter Python, make a class, export into a python object, save that object, then call VAMB with that internal object (I hope I'm not exaggerating). This is going a bit far for what could be (and what was) simply reading a file of depths, which are commonly stored in simple TSV format (either in the jgi format with variance, or even simpler). 

The requirement that users resort to a lengthier and non-generalizing process involving instantiating classes and saving them as internal objects in python is a step I hope the developers can consider revising, and bring back the simple --jgi flag, especially since there is significant lingering documentation on integrating with a Metabat2 workflow (which was part of the appeal, to me). Since the VAMB API is now defined solely based on the command-line version anyway, I propose the developers keep things simple so users can run on the commandline more flexibly, and integrate into existing workflows where assemblies were already made and depths already summarized.

**Comment by jakobnissen** [comment]
Dear @GabeAl 

I agree, it's not great the Vamb provides no command-line interface (CLI) way to pass precomputed abundances. In the upcoming Vamb v5, I have added the flag `--aemb`, which allows you to pass files in the format of `strobealign`'s --aemb output. However, truth be told, that's not great either. I don't want Vamb's parser to be tied to some idiosyncratic file format of one specific tool, neither MetaBAT nor strobealign. The JGI format is underspecified and relatively obscure. In contrast, something like the TSV format is more well-defined and simpler. So, in https://github.com/RasmussenLab/vamb/pull/356, I removed `--aemb` and added `--abundances_tsv`, which allows you to pass in a TSV file with precomputed abundances. Will that work for your use case?

W.r.t single-sample binning, you could randomly group your samples in groups of, say, 10 or 100 samples, then bin those. Even when the grouping is random, binning will be more effective than single-sample binning. I haven't run the numbers, but it's also possible that it will be faster to pool samples like this: It'll be slower in the mapping step, but faster in the binning step. If your mapper is sufficiently fast (on the order of strobealign), I would imagine it's worth it.

**Comment by jakobnissen** [comment]
> Why is this the case?

I'm not sure. Co-abundance in the multi-split workflow is not really biologically meaningful, because a contig recovered from sample A doesn't really have a depth in sample B.
My guess is that the abundances, even if they don't correspond to the real, absolute level of abundance, still provide a signal from each sample. That is, when I map reads from sample A against contigs from sample B, the number of reads mapping is dependent of some function of the abundance of a sequence in sample A and the homology between the sequence in sample A and those in sample B they map to. When you do this with N samples, you just get more signal. I also imagine (not supported by any evidence yet), that sequences that are hard to distinguish, e.g. because they have 99.9% sequence similarity, can be distinguished by the preferential mapping of some reads when there is as little as a single nucleotide difference in alignment identity to another sequence. And that this semi-rare event just happens more times with more samples, thus providing a signal.

For what it's worth, I don't think forcing 100% alignment identity matters much. First because assemblers introduce artifacts and merge sequences to similar strains to form consensus contigs. So, realistically, the sequence output by an assembler may be several % different from the original sequence.
Second, because I believe part of the signal of co-abundance comes from the aligner preferentially mapping reads to subjects with N mismatches instead of N+1 mismatches.

There is definitely more work to do in this space. Anyway, closing this issue as solved. You're welcome to open another.
